### DISCUSSION TITLE
PPO for rl_games vs skrl

### DISCUSSION THREAD

**Discussion by berttggg** [Toni-SM/skrl#103] [Q&A] [answered]
Hi, I am using the PPO for Isaac-Lift-Franka-v0, which is a task in NVIDIA Isaac Orbit. I found that the performance of PPO in rl_games is much better than skrl. So, I have tried to adjust the skrl's parameterS as same as rl_games's, but it does not work. I am wondering if I overlook something or if the architecture of PPO in rl_games and skrl are fundamentally different. Could you provide me any advice or insight about how to make the PPO in skrl performs as well as rl_games? 

Thank you in advance.

Here is the PPO parameters for skrl:

```
seed: 42

# Models are instantiated using skrl's model instantiator utility
# https://skrl.readthedocs.io/en/develop/modules/skrl.utils.model_instantiators.html
models:
  separate: False
  policy:  # see skrl.utils.model_instantiators.gaussian_model for parameter details
    clip_actions: True
    clip_log_std: False
    min_log_std: -20.0
    max_log_std: 2.0
    input_shape: "Shape.STATES"
    hiddens: [256, 128, 64]
    hidden_activation: ["elu", "elu", "elu"]
    output_shape: "Shape.ACTIONS"
    output_activation: ""
    output_scale: 1.0
  value:  # see skrl.utils.model_instantiators.deterministic_model for parameter details
    clip_actions: False
    input_shape: "Shape.STATES"
    hiddens: [256, 128, 64]
    hidden_activation: ["elu", "elu", "elu"]
    output_shape: "Shape.ONE"
    output_activation: ""
    output_scale: 1.0


# PPO agent configuration (field names are from PPO_DEFAULT_CONFIG)
# https://skrl.readthedocs.io/en/latest/modules/skrl.agents.ppo.html
agent:
  rollouts: 32
  learning_epochs: 5
  mini_batches: 16
  discount_factor: 0.99
  lambda: 0.95
  learning_rate: 5.e-4
  learning_rate_scheduler: "KLAdaptiveRL"
  learning_rate_scheduler_kwargs:
    kl_threshold: 0.008
  state_preprocessor: "RunningStandardScaler"
  state_preprocessor_kwargs: {"size": env.observation_space, "device": device}
  value_preprocessor: "RunningStandardScaler"
  value_preprocessor_kwargs: {"size": 1, "device": device}
  random_timesteps: 0
  learning_starts: 0
  grad_norm_clip: 1.0
  ratio_clip: 0.2
  value_clip: 0.2
  clip_predicted_values: True
  entropy_loss_scale: 0.0
  value_loss_scale: 4.0
  kl_threshold: 0
  # rewards_shaper_scale: 0.01
  # logging and checkpoint
  experiment:
    directory: "lift"
    experiment_name: ""
    write_interval: 120
    checkpoint_interval: 200


# Sequential trainer
# https://skrl.readthedocs.io/en/latest/modules/skrl.trainers.sequential.html
trainer:
  timesteps: 240000

```

Here is the PPO parameters for rl_games:
```

params:
  seed: 42

  # environment wrapper clipping
  env:
    clip_observations: 10.0
    clip_actions: 1.0

  algo:
    name: a2c_continuous

  model:
    name: continuous_a2c_logstd

  network:
    name: actor_critic
    separate: False
    space:
      continuous:
        mu_activation: None
        sigma_activation: None

        mu_init:
          name: default
        sigma_init:
          name: const_initializer
          val: 0
        fixed_sigma: True
    mlp:
      units: [256, 128, 64]
      # units: [512, 256, 128]
      activation: elu
      d2rl: False

      initializer:
        name: default
      regularizer:
        name: None

  load_checkpoint: False # flag which sets whether to load the checkpoint
  load_path: '' # path to the checkpoint to load

  config:
    name: lift
    env_name: rlgpu
    device: 'cuda:0'
    device_name: 'cuda:0'
    multi_gpu: False
    ppo: True
    mixed_precision: False
    normalize_input: True
    normalize_value: True
    value_bootstrap: True
    num_actors: -1
    reward_shaper:
      scale_value: 1.0
    normalize_advantage: True
    gamma: 0.99
    tau: 0.95
    learning_rate: 5e-4
    lr_schedule: adaptive
    schedule_type: legacy
    kl_threshold: 0.008
    score_to_win: 10000
    max_epochs: 10000
    save_best_after: 20
    save_frequency: 20
    print_stats: True
    grad_norm: 1.0
    entropy_coef: 0.0
    truncate_grads: True
    e_clip: 0.2
    horizon_length: 32
    minibatch_size: 2048
    mini_epochs: 5
    critic_coef: 4
    clip_value: True
    seq_lenqq: 4
    bounds_loss_coef: 0.0001

```

**Comment by Toni-SM** [comment]
Hi @berttggg 

A quick question: what maximum mean reward values are you getting with *rl_games* and *skrl*?

**Comment by Toni-SM** [comment, marked as answer]
Solution in Isaac Sim forum: https://forums.developer.nvidia.com/t/ppo-for-rl-games-vs-skrl/262682/4?u=toni.sm
